### PR TITLE
feat: add logical cpu cores and memory into grafana

### LINKED ISSUE #3821: Show the maximum available CPU/Memory on TiFlash Grafana panel
## Feature Request

**Is your feature request related to a problem? Please describe:**

Now we can not know what's the maximum available CPU/Memory on Grafana panel, especially on k8s deployed env.

**Describe the feature you'd like:**

Show the maximum available CPU/Memory on TiFlash Grafana panel.
I've created a basic branch for getting the maximum available CPU/Memory quota: https://github.com/JaySon-Huang/tics/commits/fix_cpu_count

The show style could be like this:
image

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

Useful for checking whether TiFlash resource is running out, especially on k8s env

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
Signed-off-by: Lloyd-Pottiger <yan1579196623@gmail.com>

### What problem does this PR solve?

Issue Number: close #3821 

Problem Summary:

### What is changed and how it works?
add logical cpu cores and memory into grafana

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by Lloyd-Pottiger** [comment]
looks like:
image

**Review by jiaqizho — APPROVED** [review]

File: metrics/grafana/tiflash_summary.json [hunk 1]
```diff
@@ -1117,6 +1117,142 @@
...
+            },
+            "text": {},
+            "textMode": "auto",
+            "alignmentMode": "center"
+          },
+          "pluginVersion": "7.5.7",
+          "targets": [
+            {
+              "exemplar": true,
+              "expr": "sum(tiflash_system_current_metric_LogicalCPUCores{tidb_cluster=\"$tidb_cluster\", instance=~\"$instance\"}) by (instance)",
```
Comment (JaySon-Huang) [inline metrics/grafana/tiflash_summary.json:1174]: I think there should be `k8s_cluster=\"$k8s_cluster\", ` in the `expr` field?

Comment (Lloyd-Pottiger) [reply metrics/grafana/tiflash_summary.json]: fixed!

**Comment by Lloyd-Pottiger** [comment]
now it looks like:
image

**Review by JaySon-Huang — APPROVED** [review]
LGTM for the code changes

**Comment by Lloyd-Pottiger** [comment]
put it into `CPU USAGE` panel seems confusing…
g1UaNlG2i7

File: dbms/src/Common/CurrentMetrics.cpp [hunk 2]
```diff
@@ -22,6 +22,17 @@
...
+    M(LeaderElection)                           \
+    M(EphemeralNode)                            \
+    M(DelayedInserts)                           \
+    M(ContextLockWait)                          \
+    M(StorageBufferRows)                        \
+    M(StorageBufferBytes)                       \
+    M(DictCacheRequests)                        \
+    M(Revision)                                 \
+    M(LogicalCPUCores)                          \
+    M(MemoryCapacity)                           \
```
Comment (Lloyd-Pottiger) [inline dbms/src/Common/CurrentMetrics.cpp:35]:
```suggestion
    M(LogicalCPUCores)                          \
    M(MemoryCapacity)                           \
```

File: dbms/src/Server/Server.cpp [hunk 3]
```diff
@@ -177,6 +177,13 @@ namespace
 }
 } // namespace
 
+namespace CurrentMetrics
+{
+extern const Metric Revision;
+extern const Metric LogicalCPUCores;
+extern const Metric MemoryCapacity;
```
Comment (Lloyd-Pottiger) [inline dbms/src/Server/Server.cpp:184]:
```suggestion
extern const Metric LogicalCPUCores;
extern const Metric MemoryCapacity;
```

File: dbms/src/Common/CurrentMetrics.cpp (hunk 2 above)
Comment (Lloyd-Pottiger) [inline dbms/src/Common/CurrentMetrics.cpp:35]:
```suggestion
    M(LogicalCPUCores)                          \
    M(MemoryCapacity)                           \
```

**Pull request merged**